docs/user-guide/faq.md

```diff
@@ -0,0 +1,26 @@
+# Frequently Asked Questions
+
+## How do I use a plugin not implemented in nixvim
+
+Using a plugin not supported by nixvim, but packaged in nixpkgs is straightforward:
+
+- Register the plugin through `extraPlugins`: `extraPlugins = [pkgs.vimPlugins."<plugin name>"]`.
+- Configure the plugin through `extraConfigLua`: `extraConfigLua = "require('my-plugin').setup({foo = "bar"})";`
+
+## How do I use a plugin not packaged in nixpkgs
+
+This is straightforward too, you can add the following to `extraPlugins` for a plugin hosted on GitHub:
+
+```nix
+extraPlugins = [(pkgs.vimUtils.buildVimPlugin {
+    name = "my-plugin";
+    src = pkgs.fetchFromGitHub {
+        owner = "<owner>";
+        repo = "<repo>";
+        rev = "<commit hash>";
+        hash = "<nix NAR hash>";
+    };
+})];
+```
```

Comment:
```suggestion
extraPlugins = [
  (pkgs.vimUtils.buildVimPlugin {
    name = "my-plugin";
    src = pkgs.fetchFromGitHub {
        owner = "<owner>";
        repo = "<repo>";
        rev = "<commit hash>";
        hash = "<nix NAR hash>";
    };
  })
];
```
```

Reply:
I specifically don't want to do that, in order to reduce the length of the example section.

Reply:
Ok, as you want.
Personally, I find it more readable to break a line but you get to choose :)

Reply:
I agree on readability of the code, but reducing the length of the doc will increase overall readability I think